packages/react-components/src/components/TogetherModeOverlay.tsx

```diff
@@ -0,0 +1,283 @@
+import { _pxToRem, _remToPx } from '@internal/acs-ui-common';
+/* @conditional-compile-remove(together-mode) */
+/**
+ * Signaling action overlay component props
+ *
+ * Can be used with {@link VideoTile}.
+ *
+ * @internal
+ */
+type TogetherModeParticipantStatus = {
+  reaction?: Reaction;
+  scaledSize?: number;
+  isHandRaised?: boolean;
+  isSpotlighted?: boolean;
+  isMuted?: boolean;
+  id: string;
+  seatPositionStyle: TogetherModeSeatStyle;
+  displayName: string;
+  showDisplayName: boolean;
+};
+
+/* @conditional-compile-remove(together-mode) */
+/**
+ * TogetherModeOverlay component renders an empty JSX element.
+ *
+ * @returns {JSX.Element} An empty JSX element.
+ */
+export const TogetherModeOverlay = React.memo(
+  (props: {
+    emojiSize: number;
+    reactionResources: ReactionResources;
+    localParticipant: VideoGalleryLocalParticipant;
+    remoteParticipants: VideoGalleryRemoteParticipant[];
+    togetherModeSeatPositions: VideoGalleryTogetherModeParticipantPosition;
+  }) => {
+    const locale = useLocale();
+    const theme = useTheme();
+    const callingPalette = (theme as unknown as CallingTheme).callingPalette;
+
+    const { emojiSize, reactionResources, remoteParticipants, localParticipant, togetherModeSeatPositions } = props;
+    const [visibleSignals, setVisibleSignals] = useState<{ [key: string]: TogetherModeParticipantStatus }>({});
+    const [hoveredParticipantID, setHoveredParticipantID] = useState('');
+
+    const hideSignalForParticipantsNotInTogetherMode = useCallback(() => {
+      const removedVisibleParticipants = Object.keys(visibleSignals).filter(
+        (participantId) => !togetherModeSeatPositions[participantId]
+      );
+      // Update visible signals state instead of directly mutating it
+      setVisibleSignals((prevSignals) => {
+        const newSignals = { ...prevSignals };
+        removedVisibleParticipants.forEach((participantId) => {
+          delete newSignals[participantId];
+        });
+
+        // Trigger a re-render only if changes occurred
+        const hasChanges = Object.keys(newSignals).length !== Object.keys(prevSignals).length;
+        if (hasChanges) {
+          return newSignals;
+        }
+        return prevSignals;
+      });
+    }, [visibleSignals, togetherModeSeatPositions]);
+
+    const Testing = useCallback(() => {
+      const allParticipants = [...remoteParticipants, localParticipant];
+
+      const participantsWithVideoAvailable = allParticipants.filter(
+        (p) => p.videoStream?.isAvailable && togetherModeSeatPositions[p.userId]
+      );
+      const updatedSignals = participantsWithVideoAvailable.reduce(
+        (acc: { [key: string]: TogetherModeParticipantStatus }, p: VideoGalleryLocalParticipant) => {
+          const { userId, reaction, raisedHand, spotlight, isMuted, displayName } = p;
+          const seatingPosition = togetherModeSeatPositions[userId];
+          if (seatingPosition) {
+            acc[userId] = {
```

Comment:
lets avoid using generic naming like `acc` please call it `accumulator` if thats what you are meaning here.